src/main/java/catbot/CatBot.java

```diff
@@ -0,0 +1,102 @@
+                .addCommand("bye", args -> io.cleanup())
+                .addCommand("list", args -> io.printTaskList(taskList));
+
+        // User modifying existing tasks (through IntegerPattern, and Index)
+        BiConsumer<String, Consumer<Integer>> runIfValidIndexElseIndicateError =
+                (args, lambda) -> integerPattern.ifParsableElseDefault(args,
+                        integer -> taskList.ifValidIndexElse(integer,
+                                lambda,
+                                invalidIndex -> io.indicateInvalidIndex(invalidIndex, taskList.getIndexBounds())
+                        ));
+
+                //noinspection SpellCheckingInspection
+        commands.addCommand("mark",
+                        string -> runIfValidIndexElseIndicateError.accept(string,
+                                validIndex -> {
+                                    taskList.markTask(validIndex-1);
+                                    io.printTaskModified(taskList, validIndex);
+                                }
+                        )
+                )
+                .addCommand("unmark",
+                        string -> runIfValidIndexElseIndicateError.accept(string,
+                                validIndex -> {
+                                    taskList.unmarkTask(validIndex-1);
+                                    io.printTaskModified(taskList, validIndex);
+                                }
+                        )
+                )
+                .addCommand("delete",
+                        string -> runIfValidIndexElseIndicateError.accept(string,
+                                validIndex -> io.printTaskDeleted(taskList.removeTask(validIndex-1))
+                        )
+                );
+
+        // User creating new tasks (with SlashPattern)
+
+        BiConsumer<String, BiFunction<NamedParameterMap, BiConsumer<ErrorIndicatorIo.InvalidState, NamedParameterMap>, Optional<Task>>>
+                createTaskIfValidElseWarn = (args, bifunction) -> slashPattern.ifParsableElseDefault(args,
+                        namedParameterMap -> bifunction.apply(
+                                namedParameterMap,
+                                io::indicateArgumentInvalid
+                        ).ifPresent(task -> {
+                            taskList.addTask(task);
+                            io.printTaskAdded(taskList);
+                        })
+                );
+
+        commands.addCommand("todo",
+                        args -> createTaskIfValidElseWarn.accept(args, Task.Todo::createIfValidElse)
+                )
+                .addCommand("event",
+                        args -> createTaskIfValidElseWarn.accept(args, Task.Event::createIfValidElse)
+                )
+                .addCommand("deadline",
+                        args -> createTaskIfValidElseWarn.accept(args, Task.Deadline::createIfValidElse)
+                )
+        ;
+
+    }
```

Comment:
Perhaps you could have added this to another class like CommandMap?

Reply:
I actually spent quite a bit deliberating this, but I ended up not doing it (at least for now).

## The part where I defend myself
My thought process was that I wanted to add bot-specific things in bot-specific classes; to change what commands are supported and what behaviours result from commands, my current iteration only requires that I change or even replace `CatBot` (and possibly `CatBotConsoleIO`) with minimal rewrite.
While there might be some merit in a `CatBotCommands` class that contains this method, that would leave CatBot as a small, anaemic class (I am leaning towards this being a bad thing, but I'm not too sure of myself here).
Ultimately, I had decided that it made more sense to me to have the commands be clear from the bot description, instead of its own separate class, especially since the class would otherwise be so small as to be trivial.

## The part where I stop defending myself
I definitely see an alternative that takes your considerations into account. I could rename the current file to `CatBotMain` or `CatBotEntrypoint` to reflect its explicit purpose as a small and trivial class, put this initialization as a protected static method of `CatBotCommandLoader`, or have it be the default initialization of `CatBotCommandMap`. 
I'd love to hear your thoughts on why this design might be preferable, though; or if you had a different design in mind, I'd like to chew on that as well.

Regardless, thanks for the feedback!